From 68685c79fec301bde3292a8da3db261f7799786f Mon Sep 17 00:00:00 2001
From: zrj <rimvydas.jasinskas@gmail.com>
Date: Sat, 24 Aug 2019 22:49:38 +0300
Subject: [PATCH 49/52] vkernel: Cleanup vm_offset_t.

 The pmap_kpte() is disabled code in vkernel, there are no actual
 implementation anywhere.
 The crashdump is a global in init.c, so there is no point to prototype
 it in md_var.h, just use extern in pmap.c together with rest.
---
 sys/platform/vkernel64/include/md_var.h | 3 ++-
 sys/platform/vkernel64/platform/init.c  | 1 +
 sys/platform/vkernel64/platform/pmap.c  | 1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/sys/platform/vkernel64/include/md_var.h b/sys/platform/vkernel64/include/md_var.h
index ecb57ac4f0..f859f866b4 100644
--- a/sys/platform/vkernel64/include/md_var.h
+++ b/sys/platform/vkernel64/include/md_var.h
@@ -73,7 +73,6 @@ extern	char	sigcode[];
 extern	int	szsigcode;
 extern	vpte_t	*KernelPTA;	/* NOTE: Offset for direct VA translation */
 extern	vpte_t	*KernelPTD;
-extern	vm_offset_t crashdumpmap;
 extern  int	cpu_fxsr;
 extern  pthread_t ap_tids[MAXCPU];
 
@@ -94,7 +93,9 @@ extern int	via_feature_rng;
 struct mdglobaldata;
 struct __mcontext;
 
+#if 0
 vpte_t *pmap_kpte(vm_offset_t va);
+#endif
 void cpu_gdinit (struct mdglobaldata *gd, int cpu);
 
 void cpu_heavy_restore(void);	/* cannot be called from C */
diff --git a/sys/platform/vkernel64/platform/init.c b/sys/platform/vkernel64/platform/init.c
index bd2b467f10..6cf8c60cb3 100644
--- a/sys/platform/vkernel64/platform/init.c
+++ b/sys/platform/vkernel64/platform/init.c
@@ -49,6 +49,7 @@
 #include <sys/sockio.h>
 #include <sys/sysctl.h>
 #include <sys/un.h>
+#include <vm/vm.h>
 #include <vm/vm_page.h>
 #include <vm/vm_map.h>
 #include <sys/mplock2.h>
diff --git a/sys/platform/vkernel64/platform/pmap.c b/sys/platform/vkernel64/platform/pmap.c
index c7a2bceadb..697891f0bd 100644
--- a/sys/platform/vkernel64/platform/pmap.c
+++ b/sys/platform/vkernel64/platform/pmap.c
@@ -142,6 +142,7 @@ uint64_t		KPML4phys;	/* phys addr of kernel level 4 */
 
 extern int vmm_enabled;
 extern void *vkernel_stack;
+extern vm_offset_t crashdumpmap;
 
 /*
  * Data for the pv entry allocation mechanism
-- 
2.22.0

